File: assignments/11-bitwise/src/Easy/Q16.go
package main

import (
	"fmt"
	"strings"
)

func countConsistentStrings(allowed string, words []string) int {
	count := 0
	for _, word := range words {
		co := 0
		for _, character := range word {
			if strings.Contains(allowed, string(character)) {
				co++
			}
		}
		if co == len(word) {
			count++
		}
	}
	return count
}

func main() {
	fmt.Println(countConsistentStrings("ab", []string{"ad","bd","aaab","baa","badab"}))
	fmt.Println(countConsistentStrings("ab", []string{"ad","bd","aaab","baa","badab"}))
}

File: assignments/11-bitwise/src/Easy/Q16.java
package Easy;

public class Q16 {
    public int countConsistentStrings(String allowed, String[] words) {
        int count = 0;
        boolean[] hashFreq = new boolean[26];
        for (char x : allowed.toCharArray()) {
            hashFreq[x - 'a'] = true;
        }
        for (String word : words) {
            boolean flag = false;
            for (char x : word.toCharArray()) {
                if (hashFreq[x - 'a']) {
                    flag = true;
                } else {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                count++;
            }
        }
        return count;
    }
}
